File: CTS/CTSServer/src/main/java/TransmissionUtils/AddressPhaser.java
package TransmissionUtils;

public class AddressPhaser {
    /**
     * 字符串地址转换字节流地址方法
     *
     * @param addrStr 字符串形式的地址
     * @return 字节流形式的地址
     */
    public static byte[] stringToBytes(String addrString) {
        String[] subAddrStrings = addrString.replace("\\s", "").split("\\.");
        byte[] addrBytes = new byte[4];
        for (int i = 0; i < 4; i++) {
            int temp = Integer.parseInt(subAddrStrings[i]);
            temp = temp > 127 ? (temp - 256) : temp;
            addrBytes[i] = (byte) temp;
        }
        return addrBytes;
    }

    /**
     * 字节流地址转换字符串地址方法
     *
     * @param addrBytes 字节流形式的地址
     * @return 字符串形式的地址
     */
    public static String bytesToString(byte[] addrBytes) {
        StringBuilder addrString = new StringBuilder();
        for (int i = 0; i < 4; i++) {
            int temp = Integer.parseInt(String.valueOf(addrBytes[i]));
            temp = temp < 0 ? (temp + 256) : temp;
            addrString.append(temp);
            if (i != 3)
                addrString.append(".");
        }
        return addrString.toString();
    }
}

File: CTS/AdminUser/Transmission/AddressPhaser.cs
using System.Text;

namespace AdminUser.Transmission
{
    class AddressPhaser
    {
        /// <summary>
        /// 字符串形式转换字节流形式
        /// </summary>
        /// <param name="addrStr">字符串形式的IP地址</param>
        /// <returns>字节流形式的IP地址</returns>
        public static byte[] StringToBytes(string addrString)
        {
            string[] subAddrStrings = addrString.Replace("\\s", "").Split('.');
            byte[] addrBytes = new byte[4];
            for (int i = 0; i < 4; i++)
                addrBytes[i] = byte.Parse(subAddrStrings[i]);
            return addrBytes;
        }

        /// <summary>
        /// 字节流形式转换字符串形式
        /// </summary>
        /// <param name="addrBytes">字节流形式的IP地址</param>
        /// <returns>字符串形式的IP地址</returns>
        public static string BytesToString(byte[] addrBytes)
        {
            StringBuilder addrString = new StringBuilder();
            for (int i = 0; i < 4; i++)
            {
                addrString.Append(addrBytes[i].ToString());
                if (i != 3)
                    addrString.Append(".");
            }
            return addrString.ToString();
        }
    }
}